src/logic/otp2/controller.ts

```diff
@@ -309,111 +292,85 @@ function combineAndSortFlexibleAndTransitTripPatterns(
     return sortedTripPatterns
 }
 
-export async function searchTransit(
-    params: Otp2SearchParams,
-    extraHeaders: { [key: string]: string },
-    prevQueries?: GraphqlQuery[],
-    options?: { enableTaxiSearch?: boolean },
+// Search for results in the time between the end of a search window and the first available flexible trip. This is
+// necessary as we may get a flexible trip suggestion a few days in the future. Normal transport that would be a better
+// suggestion to the traveler may be available in that time period.
+async function searchBeforeFlexible(
+    searchDate: Date,
+    arriveBy = false,
+    flexibleTripPattern: Otp2TripPattern,
+    searchParams: any,
 ): Promise<TransitTripPatterns> {
-    const { initialSearchDate, searchFilter, ...searchParams } = params
-    const { searchDate, arriveBy } = searchParams
-    const filteredModes = filterModesAndSubModes(searchFilter)
+    const searchWindow = getMinutesBetweenDates(
+        parseISO(
+            arriveBy
+                ? flexibleTripPattern.expectedEndTime
+                : flexibleTripPattern.expectedStartTime,
+        ),
+        searchDate,
+        { min: 60 },
+    )
 
-    const getTripPatternsParams = {
+    const nextSearchParams = {
         ...searchParams,
-        modes: filteredModes,
+        searchDate,
+        searchWindow,
     }
 
-    const isFirstSearchIteration = initialSearchDate === searchDate
-    const [flexibleResults, [response, initialMetadata, routingErrors]] =
-        await Promise.all([
-            isFirstSearchIteration ? searchFlexible(params) : undefined,
-            getTripPatterns(getTripPatternsParams),
-        ])
-
-    let taxiResults: TransitTripPatterns | undefined
+    const [transitResultsBeforeFlexible, metadata] = await getTripPatterns(
+        nextSearchParams,
+    )
 
-    const noStopsInRangeErrors = routingErrors.filter(
-        ({ code }) => code === RoutingErrorCode.noStopsInRange,
+    const tripPatterns = transitResultsBeforeFlexible.filter(
+        isValidTransitAlternative,
     )
 
-    if (
-        options?.enableTaxiSearch &&
-        isFirstSearchIteration &&
-        noStopsInRangeErrors.length > 0
-    ) {
-        taxiResults = await searchTaxiFrontBack(params, {
-            access: noStopsInRangeErrors.some((e) => e.inputField === 'from'),
-            egress: noStopsInRangeErrors.some((e) => e.inputField === 'to'),
-        })
+    const queries = [getTripPatternsQuery(nextSearchParams)]
+    return {
+        queries,
+        metadata,
+        tripPatterns,
     }
+}
+
+export async function searchTransitUntilMaxRetries(
```

Comment:
This used to be part of the main searchTransport, which was called recursively. However, large portions of that function would never run on the 2-n iteration, so I felt it better to move the recursive parts here.